Resolves #5 - Support export of Groups and Group Activity

zdmc23 opened the pull request.

zdmc23 commented on exports/csv.php:
```diff
@@ -73,6 +73,9 @@ function get_post_activity( $post_type ) {
 }
 $data_type = isset( $_GET['type'] ) ? esc_url_raw( wp_unslash( $_GET['type'] ) ) : '';
+$data_type = str_replace( 'https://', '', $data_type );
+$data_type = str_replace( 'http://', '', $data_type );  // http://contacts
```
@cairocoder01 in case you're wondering about this, I needed to do this bc the `$data_type` was being set to `http://contacts` or `http://group_activities` on Ln#75, and so it was falling into the `default` case of the `switch` statement each time

cairocoder01 replied:
That's because I mistakenly used the wrong escaping/sanitize function on the GET parameter on the line above. Instead of `esc_url_raw` (which expects the parameter to be a URL), change that to `sanitize_key` (see [Data Sanitization/Escaping docs](https://developer.wordpress.org/themes/theme-security/data-sanitization-escaping/))

zdmc23 replied:
@cairocoder01 `sanitize_key` will return `httpcontacts` in my dev env, so we would need to edit the switch statement (which I don't think that we would want to do)

zdmc23 replied:
Thx for sharing! I am slowly coming up to speed with WordPress best practices

cairocoder01 commented on exports/json.php:
```diff
@@ -25,20 +25,29 @@ function dt_write_log( $log ) {
 require_once( plugin_dir_path( __FILE__ ) . '../includes/data-tools.php' );
 
 $data_type = isset( $_GET['type'] ) ? esc_url_raw( wp_unslash( $_GET['type'] ) ) : '';
+$data_type = str_replace( 'https://', '', $data_type );
+$data_type = str_replace( 'http://', '', $data_type );  // http://contacts
```
I think if you update `esc_url_raw` to `sanitize_key`, you shouldn't need the string replacement. I misunderstood some of the sanitize functions when I did code linting and used the wrong one.

cairocoder01 reviewed (commented):
@zdmc23 This looks great! Thanks! 

If you can update the sanitize function as commented, that would be wonderful.

Can you also update the section in the readme for `Filter: dt_data_reporting_configurations` to add in `groups_filter` and example `data_types` for groups and group_activity?

After that, I'm happy to approve and push through

zdmc23 commented:
Hi @cairocoder01 , I've updated to use the `sanitize_key` function and I've also updated the README for `Filter: dt_data_reporting_configurations` to add in `groups_filter` and example `data_types` for groups and group_activity?  Thx for the feedback!

cairocoder01 commented:
Thanks @zdmc23! I'll get this released shortly.
